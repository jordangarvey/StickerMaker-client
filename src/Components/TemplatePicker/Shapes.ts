/**
 * Enum defining the shapes that a sticker can be
 */
enum Shapes {
	Capsule = "capsule",
	Rect = "rect",
	Square = "square"
}

export default Shapes;
