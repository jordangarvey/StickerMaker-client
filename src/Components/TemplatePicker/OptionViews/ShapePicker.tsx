import React, { FC, Fragment } from "react";
import { StyleSheet } from "react-native";

import Categories from "../Categories";
import Colours from "../Colours";
import Shapes from "../Shapes";

import Shape from "./Shape";


/**
 * Interface for the Shape Picker props
 */
interface IShapePickerProps {
	/** The currently selected shape */
	currentShape: Shapes;
	/** Callback to be fired whenever a shape option is pressed */
	onPress(category: Categories, option: Shapes): void;
	/** The currently selected colour, to use as the preview colour */
	selectedColour: Colours;
}

/**
 * Component to render the Shape Picker
 */
const ShapePicker: FC<IShapePickerProps> = (props) => (
	<Fragment>
		{
			Object.keys(Shapes).map((shape: any) => (
				<Shape
					key={Shapes[shape]}
					onPress={props.onPress}
					shape={Shapes[shape] as Shapes}
					selected={Shapes[shape] === props.currentShape}
					selectedColour={props.selectedColour}
				/>
			))
		}
	</Fragment>
);

const styles = StyleSheet.create({
	
});

export default ShapePicker;
