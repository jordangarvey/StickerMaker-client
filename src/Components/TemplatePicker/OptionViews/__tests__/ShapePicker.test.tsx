import React from "react";

import renderer from "react-test-renderer";

import Shapes from "../../Shapes";

import ShapePicker from "../ShapePicker";


describe("The Shape Picker componentâ€™s snapshot", () => {
	it("renders as expected", () => {
		function mockFunction() {
			console.log("Test")
		}

		const component = renderer.create(
			<ShapePicker
				currentShape={Shapes.Rect}
				onPress={mockFunction}
			/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
