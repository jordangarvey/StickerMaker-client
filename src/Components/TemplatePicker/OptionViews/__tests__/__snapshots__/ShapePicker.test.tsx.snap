// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Shape Picker componentâ€™s snapshot renders as expected 1`] = `
Array [
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "yellow",
        "borderRadius": 20,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <Text>
    Square
  </Text>,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "green",
        "borderColor": "green",
        "borderRadius": 20,
        "borderWidth": 3,
        "height": 50,
        "opacity": 1,
        "width": 80,
      }
    }
  />,
  <Text>
    Rectangle
  </Text>,
]
`;
